File: java/UTest.java
import java.util.Arrays;

public class UTest {
    /**
     * Holds a reference to a string array that will be later compared to another
     * string array.
     */
    public static class Expectation {
        private String[] arrayA;

        public Expectation(String[] arrayA) {
            this.arrayA = arrayA;
        }

        /**
         * Checks for equality between two string arrays.
         *
         * @param arrayB Array to be compared.
         * @throws Exception Will throw exception if arrays are not equal.
         */
        public void toEqual(String[] arrayB) throws Exception {
            if (!Arrays.equals(arrayA, arrayB)) {
                throw new Exception(String.format(
                    "\nInequality\n%s\n%s",
                    Arrays.toString(arrayA),
                    Arrays.toString(arrayB)));
            }
        }
    }

    /**
     * Factory for Expectation, used to assert in the expect().toEqual() style.
     *
     * @param arrayA Array to be compared.
     * @return Expectation instance.
     */
    public static Expectation expect(String[] arrayA) {
        return new Expectation(arrayA);
    }
}

File: javascript/utest.js
/**
 * Returns an object used to compare two arrays of strings.
 *
 * @param {string[]} arrayA Array to compare.
 * @returns {object} An object used to compare two arrays of strings.
 */
const expect = (arrayA) => ({
  /**
   * Checks for equality between two arrays of strings.
   *
   * @param {string[]} arrayB Expected array to compare.
   * @throws Will throw an error if arrays are not equal.
   */
  toEqual: (arrayB) => {
    for (let index = 0; index < arrayB.length; index++) {
      if (arrayA[index] !== arrayB[index]) {
        throw new Error(`\nInequality\n[${arrayA}]\n[${arrayB}]`)
      }
    }
  }
})

module.exports = expect
